src/client/datascience/interactive-window/nativeInteractiveWindowProvider.ts

```diff
@@ -109,69 +110,83 @@ export class NativeInteractiveWindowProvider implements IInteractiveWindowProvid
+                    this.serviceContainer.get<IConfigurationService>(IConfigurationService),
+                    this.serviceContainer.get<ICommandManager>(ICommandManager),
+                    this.serviceContainer.get<INotebookExporter>(INotebookExporter),
+                    this.serviceContainer.get<IWorkspaceService>(IWorkspaceService),
+                    resource,
+                    mode,
+                    this.serviceContainer.get<IPythonExtensionChecker>(IPythonExtensionChecker),
+                    this.serviceContainer.get<IExportDialog>(IExportDialog),
+                    notebookDocument,
+                    this.notebookControllerManager,
+                    this.kernelProvider,
+                    this.disposables,
+                    this.serviceContainer.get<IJupyterDebugger>(IJupyterDebugger)
+                );
+                this._windows.push(result);
 
-        // This is the last interactive window at the moment (as we're about to create it)
-        this.lastActiveInteractiveWindow = result;
+                // This is the last interactive window at the moment (as we're about to create it)
+                this.lastActiveInteractiveWindow = result;
 
-        // When shutting down, we fire an event
-        const handler = result.closed(this.onInteractiveWindowClosed);
-        this.disposables.push(result);
-        this.disposables.push(handler);
-        this.disposables.push(result.onDidChangeViewState(this.raiseOnDidChangeActiveInteractiveWindow.bind(this)));
+                // When shutting down, we fire an event
+                const handler = result.closed(this.onInteractiveWindowClosed);
+                this.disposables.push(result);
+                this.disposables.push(handler);
+                this.disposables.push(
+                    result.onDidChangeViewState(this.raiseOnDidChangeActiveInteractiveWindow.bind(this))
+                );
 
-        // Show in the background
-        result.show().ignoreErrors();
+                // fire created event
+                this._onDidCreateInteractiveWindow.fire(result);
+                return result;
+            });
+    }
 
-        // fire created event
-        this._onDidCreateInteractiveWindow.fire(result);
+    protected async create(resource: Resource, mode: InteractiveWindowMode): Promise<NativeInteractiveWindow> {
+        // If there's no resource, just create a new one
+        if (resource === undefined) {
+            return this.createInteractiveWindowPromise(resource, mode);
+        }
 
-        return result;
+        const existingPromise = this.mapOfResourcesToInteractiveWindowPromises.get(resource.toString());
+        // If there's a resource and no existing promise, create a new one
+        if (existingPromise === undefined) {
+            const promise = this.createInteractiveWindowPromise(resource, mode);
+            this.mapOfResourcesToInteractiveWindowPromises.set(resource.toString(), promise);
```

Comment:
Previously interactive window creation was asynchronous and could take a while, so if you ran a bunch of cells it was possible for the previous interactive window to still be in the process of initializing, and this caused us to try and create two interactive windows for the same submitting file, which uncovered a race condition in VS Code where the editor input might not have a notebook model initialized yet. This ensures that we reuse the existing init promise if a bunch of requests from the same file come in at once.